alana/prompt.py

```diff
@@ -161,7 +177,7 @@ def gen(user: Optional[str] = None, system: str = "", messages: Optional[List[Me
         **kwargs,
     )
     if append == True:
-        _append_assistant_message(messages=messages, output=output)
+        _append_assistant_message(messages=constructed_messages, output=output)
     return output.content[0].text
```

Comment:
Looks fine. Maybe set messages=None to ensure that we don't accidentally use it in place of `constructed_messages`?